pkg/codegen/hcl2/binder_schema.go

```diff
@@ -231,6 +231,8 @@ func (b *binder) schemaTypeToType(src schema.Type) (result model.Type) {
 // The result may be a *schema.UnionType if multiple schema types are associaged with the input type.
 func GetSchemaForType(t model.Type) (schema.Type, bool) {
 	switch t := t.(type) {
+	case *model.ListType:
+		return GetSchemaForType(t.ElementType)
```

Comment:
Let's change this to return an appropriate `schema.ArrayType`:

```suggestion
		if element, ok := GetSchemaForType(t.ElementType); ok {
			return &schema.ArrayType{ElementType: element}, true
		}
		return nil, false
```

Maybe do the same for `*model.MapType` while you're in here.

Reply:
FWIW, I'm not certain that this is the right approach for composite types in general. I think I'd rather be faithful to the annotations on the type, but this would require allowing all composite types to be annotated. Something to think about for the future; what we have is fine.